### DISCUSSION TITLE
Suggestion for improvement :)

### DISCUSSION THREAD

**Discussion by roy200600** [amidaware/tacticalrmm#960] [General]
I recently installed the system on a server and I am testing it for my needs
I do notice that the system is very good and also very light
There are lots of good things and also some that are missing
What I miss most personally is a customer card system for a customer group

This means that a tense customer group will enter a link which the system will build for him where he can submit a call to the service

Is in the future it in planning I do not know
But my suggestion is to create such a system and integrate it

**Comment by dinger1986** [comment]
Do you mean a chat system or a way for the customer to raise a ticket directly from the systray or something?

**Reply by roy200600** [reply]
A way for the customer to send a card to the system

**Reply by dinger1986** [reply]
Do you mean a ticket? So they can log when something has a fault?

**Reply by roy200600** [reply]
yes

**Reply by dinger1986** [reply]
Tactical doesn't have a ticketing system at the moment

**Reply by roy200600** [reply]
I know therefore I suggest inserting in the next update

**Reply by dinger1986** [reply]
A whole ticketing system?

**Reply by roy200600** [reply]
yes

**Reply by dinger1986** [reply]
thats not going to be in the next update its a fairly big thing to add. feel free to add comments to https://github.com/wh1te909/tacticalrmm/issues/253

**Reply by dkleckner498** [reply]
Is there a list of items or projects to this being worked on.  I think my biggest thing is a couple of items listed below:

Linux Agents - I know MeshCentral can do Linux but not sure where we are with the Agent side. - this one is a major for me.   (I am also sure others are probably asking for MAC)

As Roy200600 suggested, however we don't need a ticket system built into RMM.  But would like to have it where we can push out an app tray on Windows that we can customize the options on that icon like with other RMM tools.  For users to submit tickets but it uses a form based fill that you can have it on the back end send an email to whichever email that is listening for your ticket system of choice.  It would include information regarding what system it came from, the user that is currently logged in and so forth in that email body including the request.

Found a possible bug, if you know the url of the take control of the mesh window that opens.  You can basically bookmark that url and open it without actually being logged into Tactical RMM and take control of that machine.  This is actually a huge security issue.

To have basically a script store or even a check store where on top of creating your own custom checks, you can have a store that maybe community driven to go and click add to add the script check or system check to.  Also having a default set of predefined checks for systems, usually scanning the first time a system is added via the agent for all services that are running and finding common services such as SQL or other types of normal checked services that automatically adds to the check list for that server or workstation.

Eventually but most likely way down the road.  A way per a client to run a Executive Report to show all the systems and their patch and health status at the time of running.

Ability to ldap or use SSO to something like Azure or whichever service.  Usually may just take an API but would be awesome to tie in and use an SSO platform that we already have MFA built in for.

I think there are others and I can add on here soon, but for me at least this is a good start.  I appreciate all the work you all have been doing for this and really look forward to seeing some greatness come out of this project.  Would love to test development builds as well.

**Reply by bc24fl** [reply]
> 
> Found a possible bug, if you know the url of the take control of the mesh window that opens. You can basically bookmark that url and open it without actually being logged into Tactical RMM and take control of that machine. This is actually a huge security issue.
> 

From my testing, this is not the case.  You can simply copy the URL to an Incognito window to verify that it forces you to login.  Also, for future security issues/concerns, please abide by Amidaware's security guidelines here:  https://wh1te909.github.io/tacticalrmm/security/ .  

Thanks,

**Reply by dinger1986** [reply]
> Is there a list of items or projects to this being worked on.  I think my biggest thing is a couple of items listed below:
> 
> 
Yes issues/feature requests covers pretty much all of the below, feel free to add comments to each of them.
> 
> Linux Agents - I know MeshCentral can do Linux but not sure where we are with the Agent side. - this one is a major for me.   (I am also sure others are probably asking for MAC)
> 
> 
Issue #461
> 
> As Roy200600 suggested, however we don't need a ticket system built into RMM.  But would like to have it where we can push out an app tray on Windows that we can customize the options on that icon like with other RMM tools.  For users to submit tickets but it uses a form based fill that you can have it on the back end send an email to whichever email that is listening for your ticket system of choice.  It would include information regarding what system it came from, the user that is currently logged in and so forth in that email body including the request.
> 
> 
Issue #253
> 
> Found a possible bug, if you know the url of the take control of the mesh window that opens.  You can basically bookmark that url and open it without actually being logged into Tactical RMM and take control of that machine.  This is actually a huge security issue.
> 
> 
Doesn't work, I have tried this on 5 machines this morning, only works if you are logged in, please read predefined steps for reporting security bugs/issues etc. mesh does the same and generates a url for each agent which only works when logged in. @bc24fl covered this as well.
> 
> To have basically a script store or even a check store where on top of creating your own custom checks, you can have a store that maybe community driven to go and click add to add the script check or system check to.  Also having a default set of predefined checks for systems, usually scanning the first time a system is added via the agent for all services that are running and finding common services such as SQL or other types of normal checked services that automatically adds to the check list for that server or workstation.
> 
> 
Do you mean community scripts and automation? Both built into Tactical RMM since day dot. https://wh1te909.github.io/tacticalrmm/contributing_community_scripts/
> 
> Eventually but most likely way down the road.  A way per a client to run a Executive Report to show all the systems and their patch and health status at the time of running.
> 
> 
Reports are being working on Issue #508
> 
> Ability to ldap or use SSO to something like Azure or whichever service.  Usually may just take an API but would be awesome to tie in and use an SSO platform that we already have MFA built in for.
> 
> 
Issue #412
> 
> I think there are others and I can add on here soon, but for me at least this is a good start.  I appreciate all the work you all have been doing for this and really look forward to seeing some greatness come out of this project.  Would love to test development builds as well.
> 
> 

Maybe these links would be useful:

https://youtu.be/v5Wuvi_fjTI
https://wh1te909.github.io/tacticalrmm/

**Reply by wh1te909** [reply]
also the kanban board will show you what is actively being worked on https://github.com/wh1te909/tacticalrmm/projects/2

**Reply by silversword411** [reply]
Suggestion for how to suggest/do things.

GitHub discussions are for general chat, but anything on them will be lost in time because they're not tracked long-term

GitHub Issues should be single issue items. These are separated so they are one item: bug, enhancement/feature request etc.

Anyone posting an issue should search before creating a new issue to make sure it's not already there. We'll close as duplicate if someone does that.

Issues can flow into the projects kanban so you can see active dev work

If you want to work and contribute code, do the work on GitHub code, and create PRs. You can inter-reference commits, PRs and issues.
